plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -516,6 +532,31 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     }
   }
 
+  async refreshByRefreshKeys(
+    txOpaque: Transaction,
+    options: RefreshByKeyOptions,
+  ) {
+    const tx = txOpaque as Knex.Transaction;
+    const { keys } = options;
+
+    const updateResult = await tx<DbRefreshStateRow>('refresh_state')
+      .whereIn('entity_ref', function selectEntityRefs(tx2) {
+        tx2
+          .whereIn('key', keys)
+          .select({
+            entity_ref: 'refresh_keys.entity_ref',
+          })
+          .from('refresh_keys');
+      })
+      .update({ next_update_at: tx.fn.now() });
+
+    if (updateResult === 0) {
+      throw new NotFoundError(
+        `Failed to schedule ${JSON.stringify(keys)} for keys`,
+      );
```

Comment:
I don't think this should be an error tbh? Could easily have attempts to refresh with 0 hits

Reply:
Correct, this shouldn't be an error.